classes/class-ledyer-lco-gateway.php

```diff
@@ -433,6 +434,30 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 			return apply_filters( 'wc_ledyer_checkout_process_refund', false, $order_id, $amount, $reason );
 		}
 
+		/**
+		 * Update order status to "on-hold" on thank you page if not already paid.
+		 *
+		 * @param int $order_id WooCommerce order ID.
+		 */
+		public function update_status_on_thankyou( $order_id ) {
+			$order = wc_get_order( $order_id );
+			if ( empty( $order ) ) {
+				return;
+			}
+
+			$order_key = filter_input( INPUT_GET, 'key', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+			if ( empty( $order_key ) || ! hash_equals( $order->get_order_key(), $order_key ) ) {
+				return;
+			}
+
+			if ( ! empty( $order->get_date_paid() ) || 'on-hold' === $order->get_status() ) {
+				return;
+			}
+
+			$order->update_status( 'on-hold', __( 'Awaiting payment confirmation from Ledyer.', 'ledyer-checkout-for-woocommerce' ) );
```

Comment:
Issue: Lets discuss this internally before we move forward with this. We have decided before to not put it to on-hold for normal purchases since this will trigger a separate mail to customers, and in a lot of cases the processing status is set almost at the same time when the callback happens.

This also has the risk of setting the status to on-hold at the same time we are processing the order during the callback as well, which can also cause the on-hold status to be set after the processing status is set due to race conditions.